### PR TITLE
[release-2.13] Upgrade golang to 1.22.5 

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
cherry picked from commit ddea70bd87fa40ad8654154280276de631fccea6 #8600

### CODE REVIEW

**Review by ying-jeanne — APPROVED** [review]
LGTM

**Comment by ying-jeanne** [comment]
> **Building new version of mimir-build-image**. After image is built and pushed to Docker Hub, a new commit will automatically be added to this PR with new image version `grafana/mimir-build-image:pr8604-40bd216be4`. This can take up to 1 hour.

hmmm

**Comment by dimitarvdimitrov** [comment]
I interrupted the build image workflow because we're ok using the build image from `main`. The only additional diff is the version of `mixtool`

**Comment by dimitarvdimitrov** [comment]
in order to have a true build image dockerfile and image tag we're going to backport the mixtool changes to 2.13 and then continue to backport the golang version

**Comment by dimitarvdimitrov** [comment]
> in order to have a true build image dockerfile and image tag we're going to backport the mixtool changes to 2.13 and then continue to backport the golang version

apparently there are other changes, like https://github.com/grafana/mimir/pull/8385

I don't want to keep expanding the scope. So I will only udpate golang version and let CI push another build image with old skopeo, mixtool, and golangcilint

File: Makefile [hunk 1]
```diff
@@ -275,7 +275,7 @@ mimir-build-image/$(UPTODATE): mimir-build-image/*
 # All the boiler plate for building golang follows:
 SUDO := $(shell docker info >/dev/null 2>&1 || echo "sudo -E")
 BUILD_IN_CONTAINER ?= true
-LATEST_BUILD_IMAGE_TAG ?= pr8347-55c99be7fb
+LATEST_BUILD_IMAGE_TAG ?= pr8600-8a8fd767c7
```
Comment (dimitarvdimitrov) [inline Makefile:278]:
```suggestion
LATEST_BUILD_IMAGE_TAG ?= pr8347-55c99be7fb
```

let's let CI update it with a new image (hoping this comment will suffice)

**Pull request merged**